cluster-up/cluster/kind-k8s-sriov-1.14.2/config_sriov.sh

```diff
@@ -78,16 +78,18 @@ for ifs in "${sriov_pfs[@]}"; do
     continue
   fi
 
-  # These values are used to populate the network definition policy yaml.
-  # We just use the first suitable pf
-  # We need the num of vfs because if we don't set this value equals to the total, in case of mellanox
-  # the sriov operator will trigger a node reboot to update the firmware
-  export NODE_PF="$ifs_name"
-  export NODE_PF_NUM_VFS=$(cat /sys/class/net/"$NODE_PF"/device/sriov_totalvfs)
-  break
+
+  if [ -z "$NODE_PF" ]; then
```

Comment:
Why do we need this change? Is it so the value is set only during the first iteration? If so, could you either refactor the loop or at least add a comment before this line explaining it?

Reply:
> Why do we need this change? Is it so the value is set only during the first iteration? If so, could you either refactor the loop or at least add a comment before this line explaining it?

Even though it's not probably related to the CI failures, there were a lot of errors in the node configuration daemon because it was finding devices in sysfs but it wasn't find the related link in the namespace (this is because we were moving only the first device in the node's name space).
With this change, the configuration daemon will see the same things it would in a non-dockerized environment, preventing additional errors.

We pick only one PF to inject it in the network configuration as we were doing before.

Reply:
I still don't get the logic of `-z` here.

Reply:
It's to choose the first (one) PF as I wrote before:

We pick only one PF to inject it in the network configuration as we were doing before.

Reply:
Probably a deficiency in the operator if that breaks it, but I guess since we have blacklists, it's good enough for environments that rely on a PF for host connectivity. I am not thrilled but I'm ok with it.